volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -373,7 +410,44 @@ def name_as_str(self) -> str:
 class dentry(objects.StructType):
 
     def path(self) -> str:
-        return self.d_name.name_as_str()
+        """ Based on __dentry_path Linux kernel function"""
```

Comment:
How do these changes affect existing linux plugins?  I'm going to ask @atcuno to review this because I don't know these functions well enough...

Reply:
Yeah, it would be better to double-check this change didn't break anything. However, `d_name` is not the path, is just the dentry name. This function mimics the kernel [__dentry_path() ](https://elixir.bootlin.com/linux/v5.8/source/fs/d_path.c#L325) and resolves the full dentry path.